pymeasure/instruments/keithley/keithleyDMM6500.py

```diff
@@ -0,0 +1,1432 @@
+        print(dmm.voltage)
+    """
+
+    MODES = {
+        "voltage": "VOLT:DC",
+        "voltage ac": "VOLT:AC",
+        "current": "CURR:DC",
+        "current ac": "CURR:AC",
+        "resistance": "RES",
+        "resistance 4W": "FRES",
+        "diode": "DIOD",
+        "capacitance": "CAP",
+        "temperature": "TEMP",
+        "continuity": "CONT",
+        "period": "PER:VOLT",
+        "frequency": "FREQ:VOLT",
+        "voltage ratio": "VOLT:DC:RAT",
+    }
+
+    MODES_HAVE_AUTORANGE = (
+        "current",
+        "current ac",
+        "voltage",
+        "voltage ac",
+        "resistance",
+        "resistance 4W",
+        "capacitance",
+        "voltage ratio",
+    )
+
+    BOOL_MAPPINGS = {True: 1, False: 0}
+
+    channels = Instrument.MultiChannelCreator(ScannerCard2000Channel, list(range(1, 11)))
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(adapter, "Keithley DMM6500 6½-Digit Multimeter", **kwargs)
+        self.adapter.connection.read_termination = "\n"
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """With context manager, this will fully close the connection."""
+        self.adapter.close()
+
+    def close(self):
+        """Fully closes the connection"""
+        self.adapter.close()
+
+    ###########
+    # General #
+    ###########
+
+    command_set = Instrument.control(
+        "*LANG?",
+        "*LANG %s",
+        """ A string property that controls the command set, which can take the values:
+        :code:`SCPI`, :code:`TSP`, :code:`SCPI2000`, and :code:`SCPI34401`.
+        The :attr:`KeithleyDMM6500` class was designed to use :code:`SCPI` command set only.
+        """,
+        validator=strict_discrete_set,
+        values=["TSP", "SCPI", "SCPI2000", "SCPI34401"],
+    )
+    mode = Instrument.control(
+        ":SENS:FUNC?",
+        ':SENS:FUNC "%s"',
+        """ A string property that controls the configuration mode for measurements,
+        which can take the values: ``current`` (DC), ``current ac``,
+        ``voltage`` (DC),  ``voltage ac``,
+        ``resistance`` (2-wire), ``resistance 4W`` (4-wire),
+        ``diode``, ``capacitance``,
+        ``temperature``, ``continuity``,
+        ``period``, ``frequency``,
+        and ``voltage ratio``.
+        """,
+        validator=strict_discrete_set,
+        values=MODES,
+        map_values=True,
+        get_process=lambda v: v.replace('"', ""),
+    )
+
+    line_frequency = Instrument.measurement(
+        ":SYST:LFR?",
+        """ Reads the power line frequency which automatically detected while the instrument is powered on.""",
+    )
+
+    range_ = Instrument.control(
+        "{function}:RANG?",
+        "{function}:RANG %s",
+        """Control the range for the currently active mode.
+
+        For frequency and period measurements, ranging applies to
+        the signal's input voltage, not its frequency""",
+    )
+
+    autorange = Instrument.control(
```

Comment:
"Good names for boolean properties are chosen such that they could also be a yes/no question: “Is the output enabled?” -> output_enabled, display_active, etc."

[documentation](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/properties.html#boolean-properties)

Therefore:
```suggestion
    autorange_enabled = Instrument.control(
```

Reply:
Thanks for reminding. I added "_enabled" to all boolean properties.